src/main/scala/firrtl/options/StageAnnotations.scala

```diff
@@ -14,6 +17,23 @@ sealed trait StageOption { this: Annotation => }
   */
 trait Unserializable { this: Annotation => }
 
+trait HowToSerialize { this: Annotation =>
```

Comment:
this needs scaladoc

Reply:
In dev we talked about renaming this, "emission" seemed to be a good word.

What about `CustomEmission` so that you have
```
case class MyAnnotation(...) extends Annotation with CustomEmission
```
Or `CustomFileEmission` or `FileEmitable`

Reply:
I like `CustomFileEmission` for its descriptiveness, but I think `FileEmitable` wins my "vote" since it is so much shorter!

Reply:
Scaladoc added. 

For the name, I'm most in favor of maximizing descriptiveness (at the cost of characters). `CustomFileEmission` is pretty clear. I'll think a bit more about this.